devlib/module/gem5stats.py

```diff
@@ -66,6 +67,7 @@ def __init__(self, target):
         self._stats_file_path = os.path.join(target.platform.gem5_out_dir,
                                             'stats.txt')
         self.rois = {}
+        self._dump_pos_cache = {0: 0}
```

Comment:
Ah yes we can probably find a better data-structure. Actually, the order in which keys are inserted depends on how `match()` is called by the user. So we can't really assume they will be inserted in increasing order and I think this is what we need to speed-up lines 176:177 :confused:. What might be usefull is some sort of key-ordered dict here instead of a simple OrderedDict. Do you know if there is a standard python data-structure doing that ? Most likely, we need some sort of tree-based hash-map, otherwise complexity will be :-1: